mem.so/src/rvs_memtest.cpp

```diff
@@ -114,53 +99,59 @@ unsigned int error_checking(const std::string& pmsg, unsigned int blockidx)
 
-	      for (int i = 0; i < MIN(MAX_ERR_RECORD_COUNT, numOfErrors); i++){
+    for (int i = 0; i < reported_errors; i++){
 
-              msg = "[" + memdata.action_name + "] " + MODULE_NAME + " " + "ERROR: the last : " +  
-                        std::to_string(host_err_addr[i]) + " \n ";
-              rvs::lp::Log(msg, rvs::loginfo);
-	      }
+            msg = "[" + memdata.action_name + "] " + MODULE_NAME + " " +  
+                std::to_string(host_err_addr[i]) + " \n ";
+            rvs::lp::Log(msg, rvs::loginfo);
+	  }
 
-	      for (int i =0; i < MIN(MAX_ERR_RECORD_COUNT, numOfErrors); i++){
 
-              msg = "[" + memdata.action_name + "] " + MODULE_NAME + " " + "ERROR: the last : " +  
-                        " ERROR:" + std::to_string(i) + " th error, expected value=0x" +  std::to_string(host_err_expect[i]) +  
-                        "current value=0x" + std::to_string(host_err_current[i]) + "current value=0x" + std::to_string(host_err_current[i]) 
-                        + "diff=0x" +  std::to_string((host_err_expect[i] ^ host_err_current[i])) + " second_ read=0x " + 
-                       std::to_string(host_err_second_read[i]) + "expect=0x" + std::to_string(host_err_expect[i]) +  
-                       "diff with expected value=0x" +  std::to_string((host_err_expect[i] ^ host_err_second_read[i])) + "\n \n";
-              rvs::lp::Log(msg, rvs::loginfo);
-	    }
+    msg = "[" + memdata.action_name + "] " + MODULE_NAME + " " + "ERROR: the last :" + 
+              std::to_string(reported_errors) + " : error details are : \n";
+    rvs::lp::Log(msg, rvs::loginfo);
+    for (int i =0; i < reported_errors; i++){
 
+          msg = "[" + memdata.action_name + "] " + MODULE_NAME + " "  +  
+                    " ERROR:" + std::to_string(i) + " th error, expected value=0x" +  std::to_string(host_err_expect[i]) +  
+                    "current value=0x" + std::to_string(host_err_current[i]) + "current value=0x" + 
+                    std::to_string(host_err_current[i]) + 
+                    " second_ read=0x " + std::to_string(host_err_second_read[i]) +  "\n \n";
+          rvs::lp::Log(msg, rvs::loginfo);
+    }
 
-	    hipMemset(ptCntOfError, 0, sizeof(unsigned int));
-	    hipMemset((void*)&ptFailedAdress[0], 0, sizeof(unsigned long)*MAX_ERR_RECORD_COUNT);;
-	    hipMemset((void*)&ptExpectedValue[0], 0, sizeof(unsigned long)*MAX_ERR_RECORD_COUNT);;
-	    hipMemset((void*)&ptCurrentValue[0], 0, sizeof(unsigned long)*MAX_ERR_RECORD_COUNT);;
 
-	    hipDeviceReset();
-	    exit(ERR_BAD_STATE);
-    }
+    hipMemset(ptCntOfError, 0, sizeof(unsigned int));
+    hipMemset((void*)&ptFailedAdress[0], 0, sizeof(unsigned long)*MAX_ERR_RECORD_COUNT);;
+    hipMemset((void*)&ptExpectedValue[0], 0, sizeof(unsigned long)*MAX_ERR_RECORD_COUNT);;
+	  hipMemset((void*)&ptCurrentValue[0], 0, sizeof(unsigned long)*MAX_ERR_RECORD_COUNT);;
+
+    hipDeviceReset();
+    exit(ERR_BAD_STATE);
 
-    return numOfErrors;
```

Comment:
**return** statement missing.

Reply:
Line 107 actually returns if no errors and in case errors we are calling exit(ERR_BAD_STATE). This is same as earlier case but re-organized code for clarity.